### ISSUE TITLE
Add automated system tests to help admins verify the installation and configuration of OnDemand

### ISSUE THREAD

**Issue by ericfranz** [OSC/ondemand#790]
smgallo wrote:

> While the documentation for the various components does explain how to use or access the component, there does not appear to be an automated test suite or manual process specified for each component to verify that the component is installed correctly (note that some components do instruct the installer to manually verify). This is a complex software and infrastructure package and it would be useful for each component to have a simple verification script that could perhaps test for the existence of a file or test the data returned from a REST endpoint. I can certainly appreciate the effort involved in creating and maintaining tests of this nature.

Internally OSC has discussed 2 wanting to add system tests for

1. Job Adapters - we would add these tests to https://github.com/OSC/ood_core and would provide the path to a cluster config, which would be used to submit some jobs, check status of them, output the status of all jobs, and display a report which would be helpful for adapter development and debugging.
1. Use automated system tests to create an "installation/environment dependency checklist" for Interactive Apps. I imagine from the dashboard, being able to submit a single batch job that would check for the existence of multiple dependencies for interactive apps, or maybe each interactive app having a "verification job" that runs, verifies it can start the interactive app, then closes with a user readable report available in the dashboard.

But we had not discussed having a set of system tests to verify different parts of OnDemand installation across the board.

smgallo what parts of the OnDemand installation process would you like to verify using automated tests?

1. installation of infrastructure
1. configuration of authentication
1. installation of core utility apps (shell, files, dashboard)
1. installation of core job apps (myjobs, activejobs)
1. other?

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201735133575781/1201737592261413) by [Unito](https://www.unito.io)

**Comment by matt257** [comment]
reviewed,
too difficult given constraints,
maybe rephrase if we want something similar later
